nengo/transforms.py

```diff
@@ -135,11 +119,13 @@ class Dense(Transform):
     def __init__(self, shape, init=1.0):
         super(Dense, self).__init__()
 
-        self.shape = shape
-
         if is_array_like(init):
             init = np.asarray(init)
 
+            # replace any unspecified shape values
+            shape = tuple(s if s != '*' else init.shape[k] if k < init.ndim
+                          else 1 for k, s in enumerate(shape))
```

Comment:
I'm OK with having this in, but since it's not actually being used in the code we should add a test (so we know it's working).